src/Slot-Examples/AbstractInitializedClassVariable.class.st

```diff
@@ -0,0 +1,40 @@
+"
+Intialized Class Variables define a default value as part of their definition.
+
+The default parameter can be literal or a block.
+
+There are two subclasses: setting the default value on class creation and lazy initialization on read.
+"
+Class {
+	#name : #AbstractInitializedClassVariable,
+	#superclass : #LiteralVariable,
+	#instVars : [
+		'default',
+		'isBlock'
+	],
+	#category : #'Slot-Examples-ClassVariables'
+}
+
+{ #category : #accessing }
+AbstractInitializedClassVariable >> default: anObject [
+	default := anObject.
+	isBlock := false
```

Comment:
isBlock variable looks redundant.
If there is anyway isBlock ifTrue logic then just ask "default isBlock"

Reply:
I for now did it in a way that you can specify a block as the default, but not evaluate it.
AbstractInitializedClassVariable default: []
vs
AbstractInitializedClassVariable defaultBlock: []

in the second case the block is evaluated, in the first not

I was trying to do it by always wrapping in a block, but that is not nice either.

Reply:
But I think it is possible to simplify: just have default:, and if people want a block they have to do [[]] 

I will try that...

Reply:
Can’t default be 

default: anObject
    self defaultBlock: [ anObject ]

Or the other way around. I am on my phone and it’s not easy to read the code.